lib/TextToImage/TextToImageProvider.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+// SPDX-FileCopyrightText: Julien Veyssier <julien-nc@posteo.net>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+namespace OCA\Replicate\TextToImage;
+
+use OCA\Replicate\AppInfo\Application;
+use OCA\Replicate\Service\ReplicateAPIService;
+use OCP\Http\Client\IClientService;
+use OCP\IL10N;
+use OCP\TextToImage\IProvider;
+use Psr\Log\LoggerInterface;
+
+class TextToImageProvider implements IProvider {
+
+	public function __construct(
+		private ReplicateAPIService $replicateAPIService,
+		private LoggerInterface $logger,
+		private IL10N $l,
+		private IClientService $clientService,
+		private ?string $userId,
+	) {
+	}
+
+	public function getId(): string {
+		return Application::APP_ID . '_image_generation';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getName(): string {
+		return $this->l->t('Replicate\'s stable diffusion Text-To-Image');
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function generate(string $prompt, array $resources): void {
+		$nbOutputs = count($resources);
+		$waitingForAlready = 0;
+		$maxWaitTime = 60 * 60;
+		try {
+			$prediction = $this->replicateAPIService->createImagePrediction($prompt, $this->userId, $nbOutputs, '768x768');
```

Comment:
Can we make this customizable?

Most modern models outputs not only square 768x768 and 1024x1024, but 1536x864,  2048x1152, 1024x576 for wide screens..

Reply:
Not immediately in this PR, long term, sure

Reply:
It's kind of needed to add parameters. As parameters are different for each models and we can't predict them. It's now possible to set any extra parameter. Up to the NC admins to read model-specific docs on Replicate's website. These extra params are global and apply to all users for tasks using the Replicate image/text generation provider.

It's also possible to choose a model by name or version code. Up to the admins to explore Replicate's website to find models.

Reply:
Yeah, sorry about the ambiguous answer